src/pilot/ovs_dpdk_setup.py

```diff
@@ -0,0 +1,146 @@
+        'sed -i "s|changeme2|' + nics[1] + '|" ' + config_file,
+        'sed -i "s|changeme3|' + nics[2] + '|" ' + config_file,
+        'sed -i "s|changeme4|' + nics[3] + '|" ' + config_file,
+        'sed -i "s|Compute::Net::SoftwareConfig:.*|' +
+        'Compute::Net::SoftwareConfig: ' + config_file + '|" ' + env_file,
+        'sed -i "s|DpdkInterfaces:.*|DpdkInterfaces: ' +
+        interfaces + '|" ' + env_file,
+        ]
+    LOG.info("Running sed commands to configure environment.")
+    for cmd in cmds:
+        LOG.info(cmd)
+        out = subprocess.Popen(cmd, shell=True, stdout=subprocess.PIPE)
+        LOG.info("output: " + str(out.stdout.read()))
+
+
+def main():
+
+    R630_NICS = ['p2p1', 'p3p1', 'p2p2', 'p3p2']
+    R730_NICS = ['p4p1', 'p5p1', 'p4p2', 'p5p2']
+    parser = argparse.ArgumentParser()
+    parser.add_argument("--env_file",
+                        dest="env_file",
+                        required=True,
+                        help="OVS DPDK Environment file")
+    parser.add_argument("--nic_config_file",
+                        dest="config_file",
+                        required=True,
+                        help="OVS-DPDK Compute NIC configs")
+    parser.add_argument("--mode",
+                        dest="mode",
+                        type=int,
+                        required=True,
+                        help="OVS-DPDK Mode")
+    LoggingHelper.add_argument(parser)
+    args = parser.parse_args()
+
+    root_logger = logging.getLogger()
+    root_logger.setLevel(args.logging_level)
+    urllib3_logger = logging.getLogger("requests.packages.urllib3")
+    urllib3_logger.setLevel(logging.WARN)
+
+    ironic_client = get_ironic_client()
+    nodes = get_ironic_nodes(ironic_client)
+
+    compute_nodes_model = set()
+    for node in nodes:
+        node = node.to_dict()
+        if 'compute' in node['properties']['capabilities']:
+            model = node['properties']['model']
+            compute_nodes_model.add(model)
+            LOG.info("Node: " + node['uuid'] + " model: " + model)
+    if len(compute_nodes_model) == 0:
+        raise Exception('ERROR: No server model found for compute nodes.')
+    elif len(compute_nodes_model) != 1:
+        raise Exception('ERROR: Multiple server models found for compute '
+                        'nodes. All compute nodes must have same '
+                        'server models.')
+
+    compute_nodes_model = list(compute_nodes_model)
+    if compute_nodes_model[0] == 'PowerEdge R630':
+        nics = R630_NICS
+    else:
+        nics = R730_NICS
+    mode = args.mode
+    if mode == 1 or mode == 2:
+        pass
+    else:
+        raise Exception('ERROR: Mode ' + mode + ' is not supported.'
+                        'Only supported values are 1 and 2.')
+    setup_ovs_dpdk_environment(nics, args.config_file, args.env_file, mode)
+
+if __name__ == "__main__":
+    try:
+        main()
+    except Exception as e:
+        print str(e)

```

Comment:
The above print statement should be replaced with a logging statement.

Reply:
Agree. Done